src/wrapper/objects/auto_byte_array.rs

```diff
@@ -0,0 +1,115 @@
+use std::mem;
+
+use jni_sys::{jbyte, JNI_ABORT};
+use log::debug;
+
+use crate::{objects::JObject, JNIEnv};
+
+/// ReleaseMode
+///
+/// This defines the release mode of AutoByteArray (and AutoPrimitiveArray) resources, and
+/// related release array functions.
+#[derive(Clone, Copy)]
+pub enum ReleaseMode {
+    /// Copy back the content and free the elems buffer.
+    Copy = 0,
+    /// Free the buffer without copying back the possible changes.
+    NoCopy = JNI_ABORT as isize,
+}
+
+/// Auto-release wrapper for pointer-based byte arrays.
+///
+/// This wrapper is used to wrap pointers returned by get_byte_array_elements.
+///
+/// These arrays need to be released through a call to release_byte_array_elements.
+/// This wrapper provides automatic array release when it goes out of scope.
+pub struct AutoByteArray<'a: 'b, 'b> {
+    obj: JObject<'a>,
+    ptr: *mut jbyte,
+    mode: ReleaseMode,
+    is_copy: bool,
+    env: &'b JNIEnv<'a>,
+}
+
+impl<'a, 'b> AutoByteArray<'a, 'b> {
+    /// Creates a new auto-release wrapper for a pointer-based byte array
+    ///
+    /// Once this wrapper goes out of scope, `release_byte_array_elements` will be
+    /// called on the object. While wrapped, the object can be accessed via
+    /// the `Deref` impl.
+    pub fn new(
+        env: &'b JNIEnv<'a>,
+        obj: JObject<'a>,
+        ptr: *mut jbyte,
+        mode: ReleaseMode,
+        is_copy: bool,
+    ) -> Self {
+        AutoByteArray {
+            obj,
+            ptr,
+            mode,
+            is_copy,
+            env,
+        }
+    }
+
+    /// Forget the wrapper, returning the original pointer.
+    ///
+    /// This prevents `release_byte_array_elements` from being called when the `AutoArray`
+    /// gets dropped. You must remember to release the array manually.
+    pub fn forget(self) -> *mut jbyte {
+        let ptr = self.ptr;
+        mem::forget(self);
+        ptr
+    }
```

Comment:
Is there a use-case when this is needed? I.e., is there a reason to use AutoArray instead of a pair of get/release when you also `forget` it?

Reply:
Not really. It's there just because I copied it from `auto_local`. I'll happily remove it.